src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntityBsonSupport.scala

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.dsde.agora.server.model
+
+import com.mongodb.DBObject
+import org.joda.time.DateTime
+import org.joda.time.format.ISODateTimeFormat
+
+
+object AgoraEntityBsonSupport {
+
+  def read(obj: DBObject): AgoraEntity =
+    AgoraEntity(
+      namespace = getOrNone(obj, "namespace"),
+      name = getOrNone(obj, "name"),
+      snapshotId = getOrNone(obj, "snapshotId"),
+      snapshotComment = getOrNone(obj, "snapshotComment"),
+      synopsis = getOrNone(obj, "synopsis"),
+      documentation = getOrNone(obj, "documentation"),
+      owner = getOrNone(obj, "owner"), // Not in DB but checked by tests. Just rolling with this for now.
+      createDate = dateOrNone(obj, "createDate"),
+      payload = getOrNone(obj, "payload"),
+      payloadObject = None, // Not in DB
+      url = None, // Not in DB
+      entityType = entityType(obj),
+      id = getOrNone(obj, "_id"),
+      methodId = getOrNone(obj, "methodId"), // Configs only
+      method = None, // Not in DB
+      managers = Seq.empty, // Not in DB
+      public = None, // Not in DB
+    )
+
+  private def getOrNone[A](obj: DBObject, key: String): Option[A] = {
+    Option(obj.get(key).asInstanceOf[A])
```

Comment:
Sorry, I should have edited my earlier reply after the discussion. To avoid a bug shown [here](https://scastie.scala-lang.org/kshakir/Y6ktNXzvTs2CfsHbB28oeg/4), this should be:

```scala
Option(obj.get(key)).map(_.asInstanceOf[A])
```